File: processor/src/test/resources/scenarios/converters/NonAccessibleConverter.java
package scenarios.converters;

import org.springframework.core.convert.converter.Converter;
import org.springframework.data.convert.ReadingConverter;

/**
 * Holds a private nested {@link Converter} that cannot be reached from any package.
 * Used to verify the visibility check rejects unconditionally inaccessible classes.
 */
public final class NonAccessibleConverter {

    private NonAccessibleConverter() {
    }

    @ReadingConverter
    private static final class Inner implements Converter<String, String> {

        public Inner() {
        }

        @Override
        public String convert(String source) {
            return source;
        }
    }
}

File: ksp/src/test/resources/scenarios/converters/NonAccessibleConverter.kt
package scenarios.converters

import org.springframework.core.convert.converter.Converter
import org.springframework.data.convert.ReadingConverter

/**
 * Top-level `private` (file-scoped) class. Cannot be referenced from any other file,
 * regardless of module or package. Used to verify that the visibility check rejects
 * unconditionally inaccessible classes.
 */
@ReadingConverter
private class NonAccessibleConverter : Converter<String, String> {
    override fun convert(source: String): String = source
}
